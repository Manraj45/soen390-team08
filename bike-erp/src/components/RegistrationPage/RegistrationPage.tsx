import { Grid, MenuItem, TextField, Typography, Select, InputLabel, FormControl, Button } from '@material-ui/core'
import axios from 'axios'
import React, { useEffect, useState } from 'react'
import bike_logo from '../../assets/images/login_bike_logo.png'
import { BACKEND_URL } from '../../core/utils/config'
import useStyles from './RegistrationPageStyle'
import { useHistory } from "react-router-dom";

interface RegistrationData {
    email: string
    organization: string
    password: string
    firstName: string
    lastName: string
    role: string
    recovery_question1: string
    recovery_question2: string
    recovery_question1_answer: string
    recovery_question2_answer: string
}

const RegistrationPage = () => {
    // Object for styling
    const classes = useStyles();

    const url = BACKEND_URL

    const [recoveryQuestions, setRecoveryQuestions] = useState({})
    const [question1, setQuestion1] = useState('')
    const [question2, setQuestion2] = useState('')

    const [registrationErrorMessage, setRegistrationErrorMessage] = useState("")

    // Hook for redirecting
    const history = useHistory();

    useEffect(() => {
        axios.get(`${url}/register/recoveryQuestion`).then(response => {
            setRecoveryQuestions(response.data)
        }).catch((error) => {
            console.log(error.data)
        })
    }, [url])

    // Handle typing of question 1
    const handleQuestion1 = (event: React.ChangeEvent<{ value: unknown }>) => {
        setQuestion1(event.target.value as string)
    }

    // Handle typing of question 2
    const handleQuestion2 = (event: React.ChangeEvent<{ value: unknown }>) => {
        setQuestion2(event.target.value as string)
    }

    const handleRegistration = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault()
        const registrationData: RegistrationData = {
            email: event.currentTarget.email.value,
            organization: event.currentTarget.organization.value,
            password: event.currentTarget.password.value,
            firstName: event.currentTarget.firstName.value,
            lastName: event.currentTarget.lastName.value,
            role: "ADMIN",
            recovery_question1: question1,
            recovery_question2: question2,
            recovery_question1_answer: event.currentTarget.answer1.value,
            recovery_question2_answer: event.currentTarget.answer2.value
        }

        axios.post(`${url}/register/submission`, registrationData).then(response => {
            history.push("/login")
        }).catch(error => {
            setRegistrationErrorMessage(error.response.data.message)
        })
    }

    return (
        <div>
            <Grid container spacing={0} direction="row" className={classes.registrationPageWrapper}>
                <Grid item xs={12} md={7} className={classes.grid}>
                    <form autoComplete="off" onSubmit={handleRegistration}>
                        <TextField name="email" label="Email" className={classes.textfield} ></TextField>

                        <br />

                        <TextField name="organization" label="Organization" className={classes.textfield}></TextField>

                        <br />

                        <TextField type="password" name="password" label="Password" className={classes.textfield}></TextField>

                        <br />

                        <div>
                            <TextField name="firstName" label="First Name" className={classes.firstName} ></TextField>
                            <TextField name="lastName" label="Last Name" className={classes.lastName} ></TextField>
                        </div>

                        <br />
                        <FormControl className={classes.recoveryQuestion}>
                            <InputLabel id="demo-simple-select-filled-label">Recovery Question 1</InputLabel>
                            <Select labelId="demo-simple-select-filled-label" id="demo-simple-select-filled" value={question1} onChange={handleQuestion1}>
                                {
                                    Object.values(recoveryQuestions).map((question: any) => (
                                        <MenuItem key={question} value={question}>{question}</MenuItem>
                                    ))
                                }
                            </Select>
                        </FormControl>

                        <br />

                        <TextField name="answer1" label="Answer Question 1" className={classes.textfield}></TextField>

                        <br />

                        <FormControl className={classes.recoveryQuestion}>
                            <InputLabel id="demo-simple-select-helper-label">Recovery Question 2</InputLabel>
                            <Select labelId="demo-simple-select-helper-label" id="demo-simple-select-helper" value={question2} onChange={handleQuestion2}>
                                {
                                    Object.values(recoveryQuestions).map((question: any) => (
                                        <MenuItem key={question} value={question}>{question}</MenuItem>
                                    ))
                                }
                            </Select>
                        </FormControl>
                        <br />
                        <TextField name="answer2" label="Answer Question 2" className={classes.textfield}></TextField>
                        <br />
                        {
                            registrationErrorMessage ? <Typography className={classes.error}>{registrationErrorMessage}</Typography>:<></>
                        }
                        <Button variant="contained" color="primary" className={classes.button} type="submit">Register</Button>
                    </form>
                </Grid>
                <Grid item xs={12} md={5} className={` ${classes.grid}`}>
                    <Typography variant="h3">Badob Inc</Typography>
                    <img src={bike_logo} alt="bike_logo" className={classes.logo}></img>
                </Grid>
            </Grid>
        </div>
    )
}

export default RegistrationPage