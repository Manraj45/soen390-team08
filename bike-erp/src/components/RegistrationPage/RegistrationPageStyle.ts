import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
    logo: {
        width: "90%"
    },
    grid: {
        [theme.breakpoints.down('sm')]: {
            marginTop: "50px"
        },
        [theme.breakpoints.up('md')]: {
            marginTop: "10px"
        }
    },
    registrationPageWrapper: {
        [theme.breakpoints.down('sm')]: {
            margin: "0",
            position: "absolute",
            top: "10%"
        },
        [theme.breakpoints.up('md')]: {
            margin: "0",
            position: "absolute",
            top: "30%"
        }
    },
    textfield: {
        width: "60%",
        height: "100%",
        marginBottom: "10px"
    },
    recoveryQuestion: {
        width: "60%",
        height: "100%",
        marginBottom: "10px"
    },
    recoveryLabel: {
        width: "20%",
        marginLeft: "155px"
    },
    firstName: {
        width: "29%",
        marginRight: "5px"
    },
    lastName: {
        width: "30%",
        marginRight: "5px"
    },
    button: {
        width: "60%",
        marginTop: "10px"
    },
    error: {
        color: "red"
    }
}))

export default useStyles