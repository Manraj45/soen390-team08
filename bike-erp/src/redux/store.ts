import { applyMiddleware, createStore } from "redux";
import rootReducer from "./rootReducer";
import thunk from "redux-thunk";

// Create store and apply thunk for API request
const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
