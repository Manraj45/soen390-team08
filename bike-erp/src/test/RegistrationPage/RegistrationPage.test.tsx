import React from 'react'
import RegistrationPage from '../../components/RegistrationPage/RegistrationPage'
import { shallow, configure, mount } from 'enzyme'
import store from '../../redux/store'
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() });

const setUp = () => {
    const wrapper = shallow(<RegistrationPage />)
    return wrapper
}

describe('LoginPage Component', () => {
    let wrapper: any;

    beforeEach(() => {
        wrapper = setUp()
    })

    it("render registration page without crashing", () => {
        const component = wrapper.find('#registrationPage')
        expect(component.length).toBe(1)
    })

    it('match rendered component with snapshot',()=>{
        const tree = renderer.create(<RegistrationPage/>).toJSON();
        expect(tree).toMatchSnapshot()
    })
})