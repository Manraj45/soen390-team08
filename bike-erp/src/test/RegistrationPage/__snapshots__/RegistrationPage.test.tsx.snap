// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Component match rendered component with snapshot 1`] = `
<div
  id="registrationPage"
>
  <div
    className="MuiGrid-root makeStyles-registrationPageWrapper-3 MuiGrid-container"
  >
    <div
      className="MuiGrid-root makeStyles-grid-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
    >
      <form
        autoComplete="off"
        onSubmit={[Function]}
      >
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-textfield-4"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            Email
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
        </div>
        <br />
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-textfield-4"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            Organization
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              name="organization"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
        </div>
        <br />
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-textfield-4"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            Password
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
            />
          </div>
        </div>
        <br />
        <div>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-firstName-7"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
            >
              First Name
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                name="firstName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-lastName-8"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
            >
              Last Name
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                name="lastName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
            </div>
          </div>
        </div>
        <br />
        <div
          className="MuiFormControl-root makeStyles-recoveryQuestion-5"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            id="demo-simple-select-filled-label"
          >
            Recovery Question 1
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-filled-label demo-simple-select-filled"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="demo-simple-select-filled"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value=""
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <br />
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-textfield-4"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            Answer Question 1
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              name="answer1"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
        </div>
        <br />
        <div
          className="MuiFormControl-root makeStyles-recoveryQuestion-5"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            id="demo-simple-select-helper-label"
          >
            Recovery Question 2
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-helper-label demo-simple-select-helper"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="demo-simple-select-helper"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value=""
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <br />
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-textfield-4"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
          >
            Answer Question 2
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              name="answer2"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
          </div>
        </div>
        <br />
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-9 MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Register
          </span>
        </button>
      </form>
    </div>
    <div
      className="MuiGrid-root  makeStyles-grid-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
    >
      <h3
        className="MuiTypography-root MuiTypography-h3"
      >
        Badob Inc
      </h3>
      <img
        alt="bike_logo"
        className="makeStyles-logo-1"
        src="login_bike_logo.png"
      />
    </div>
  </div>
</div>
`;
